tests/fs-test.rs

```diff
@@ -40,66 +56,147 @@ fn new_file_test() {
 }
 
 #[test]
-fn uhyvefilemap_test() {
+fn uhyvefilemap_non_present_file_test() {
 	env_logger::try_init().ok();
-	let output_path = PathBuf::from("foo.txt");
-	remove_file_if_exists(&output_path);
-	let bin_path = build_hermit_bin("create_file");
+	let fixture_path = get_fixture_path();
+	let mut host_path = fixture_path.clone();
+	host_path.push("file_to_write.txt");
+	remove_file_if_exists(&host_path);
+
+	let uhyvefilemap_params = [format!(
+		"{}:{}",
+		host_path.to_str().unwrap(),
+		"/root/dir/wrong.txt"
+	)]
+	.to_vec();
 
-	let mut params = Params {
+	let params = Params {
 		cpu_count: 2.try_into().unwrap(),
 		memory_size: Byte::from_u64_with_unit(64, Unit::MiB)
 			.unwrap()
 			.try_into()
 			.unwrap(),
-		file_mapping: vec!["/root/foo.txt:wrong.txt".to_string()],
+		file_mapping: uhyvefilemap_params,
 		output: Output::Buffer,
 		stats: true,
 		..Default::default()
 	};
 
 	// The file should not exist on the host OS.
-	let mut vm = UhyveVm::new(bin_path.clone(), params.clone()).unwrap();
-	let mut res = vm.run(None);
-	check_result(&res);
-	assert!(!output_path.exists());
+	let bin_path = build_hermit_bin("create_file");
+	let vm = UhyveVm::new(bin_path.clone(), params.clone()).unwrap();
+	let res = vm.run(None);
+	assert_eq!(res.code, 0);
+	assert!(!host_path.exists());
+}
+
+#[test]
+fn uhyvefilemap_present_file_test() {
+	env_logger::try_init().ok();
+	let fixture_path = get_fixture_path();
+	let mut host_path = fixture_path.clone();
+	host_path.push("file_to_write.txt");
+	remove_file_if_exists(&host_path);
+
+	let uhyvefilemap_params = [format!(
+		"{}:{}",
+		host_path.to_str().unwrap(),
+		"/root/foo.txt"
+	)]
+	.to_vec();
+
+	let params = Params {
+		cpu_count: 2.try_into().unwrap(),
+		memory_size: Byte::from_u64_with_unit(64, Unit::MiB)
+			.unwrap()
+			.try_into()
+			.unwrap(),
+		file_mapping: uhyvefilemap_params,
+		output: Output::Buffer,
+		stats: true,
+		..Default::default()
+	};
 
-	params.file_mapping = vec!["foo.txt:/root/foo.txt".to_string()];
-	vm = UhyveVm::new(bin_path, params).unwrap();
-	res = vm.run(None);
+	let bin_path = build_hermit_bin("create_file");
+	let vm = UhyveVm::new(bin_path.clone(), params.clone()).unwrap();
+	let res = vm.run(None);
 	check_result(&res);
-	verify_file_equals(&output_path, "Hello, world!");
+	verify_file_equals(&host_path, "Hello, world!");
 }
 
+/// This might break because of a misconfiguration in Landlock.
 #[test]
-fn remove_file_test() {
+fn remove_present_mapped_file_test() {
```

Comment:
This *does* break for some weird godforsaken reason, whereas `remove_present_file_with_mapped_parent_test` below does not.

This does not make a lot of sense, given that `access_all` (aka. both read and write, the set of permissions that `file_to_remove.txt` is restricted under) is supposed to include `RemoveFile` as well. In Landlock AFAIK, remove is supposed to work as long as it fails the path afterwards - and I believe that this previously did work, which is why I made the second commit that removes the path from UhyveFileMap respectively.

https://github.com/landlock-lsm/rust-landlock/blob/aaccff53ecb0fb3c4d2e8125bc9e8913fbf92424/src/fs.rs#L121-L143

Reply:
Even weirder: The `remove_present_file_with_mapped_parent_test` works locally, but it fails in the CI.

Reply:
> This does not make a lot of sense, given that access_all (aka. both read and write, the set of permissions that file_to_remove.txt is restricted under)

Answer: `RemoveFile` can only be applied to a directory, not a file. The directory's contents (aka. files) will then be removable in a transitive manner, but not the directory itself.